### ISSUE TITLE
RELEASE RP-1 v0.1 for KSP 1.3.1

### ISSUE THREAD

**Issue by Bornholio** [KSP-RO/RP-0#866]
Shall we release current developmental RP-1 to 1.3.1 and get ready for further changes towards KSP 1.4.X?

Any objections/seconds/insights?
@pjf @jwvanderbeck @ec429 @lamont-granquist @NathanKell @raidernick @rsparkyc @pap1723 @PhineasFreak @stratochief66 @aw1621107 and anyone else contributing.

I'd love to see state switching for several of the functions RP-1 is adding. Specifically tooling, training and roll out multiples for tags. regardless i'd love functional RP-1 released.  I've had a ton of fun testing it even with its grind.

**Comment by Schnobs** [comment]
Ack on all counts. It works well enough to be fun to play with.

I, too, would like to see tooling and training toggleable. However, about tooling, the "Just tool it all, fer crissakes" button which awang made turned out to be a a huge quality of life improvement. It should definitely in there: for me it makes all the difference between tooling being a gameplay consideration rather than a nuisance.

**Comment by pap1723** [comment]
What features would be included in the release and what would not be
included?

**Comment by aw1621107** [comment]
I would vote to include at least tooling, since IMNSHO it seems like that that would be by far the most significant change from the regular KSP campaign, in terms of effect on planning and costs. The "tool it all" button should also be added, but perhaps with a tweaked name to be a bit less inside joke-y.

Rollout cost multipliers are pretty important for balance, too, but AFAIK there isn't a good way for the player to see the effect of those multipliers on rollout costs, so maybe those should wait. Having some sort of feedback would be nice, so the player could determine whether that fancier engine is worth the extra rollout costs without having to do the math by hand. We should also probably make sure that the rollout cost multipliers actually make the player think about whether it's worth it, otherwise it's basically a background feature that players don't need to be aware of.

Maintenance costs seemed basically ignorable last time I played, so I'd say leave it out for now, until players get used to the more significant changes.

I haven't used training with the bugfixes yet, so I can't really say much about that. Perhaps with additional quality-of-life improvements (e.g. KAC alarms, reminders to start training x days before the intended vessel is complete, automatically starting training, etc.) it'd be easier to use for more players. Of course, that's just based on my experience of warping to vessel completion only to realize that I forgot to start training. I don't know how frequently that occurs.

Haven't used crew science, either, so can't give an opinion on that.

I can try to look into what it would take to make tooling and training toggleable, but I can't promise to get to it before the upcoming weekend. We also need to figure out what to do about corner cases (e.g. turn tooling off, start building vessel, turn tooling on. Should the build time/cost stay the original values? That seems like an easy-to-use exploit. Should the build time/cost increase? That could be a nasty surprise. And what should happen if the cost increase puts the player into the red?).

Basically whether most of RP-1 is releaseable depends on how friendly we try to be to players, I guess.

**Comment by Rockstar04** [comment]
I just started a playthough of RP-1, and trying to figure out the training for proficiency/missions was rather rough. I dont know if it would be feasible or possible to display an alert when trying to launch a craft from the KCT window, to alert the player that no kerbals are properly trained for the current pod. Auto KAC alarms and possibly even a contract get a kerbal trained once you unlock your first manned pod would be great to help bring up training since you would have just been building and launching unmanned rockets without thinking about that extra step so far in your career.

**Comment by pap1723** [comment]
To me, as of right now, the only feature that is straightforward, makes
sense, and has a good player UI system is tooling. I personally think
rollout cost modifiers, maintenance costs and crew training should be
removed until such time that NathanKell returns to fully implement them
properly for users. I know from my playthroughs, even though I was
integrated in the development, these features were not (and are not) ready
for a proper release to a user base that does not have a good way of seeing
the information at all times.

As far as the science stuff, I implemented it, so I cannot comment on how
it is used. I thought Bornholio did a really good job of writing a Wiki to
explain those features, and they are not difficult to understand generally.

**Comment by pap1723** [comment]
Sorry, I hit submit too soon.

By removing the rollout costs, astronaut training and maintenance, contract
payouts are going to have to be completely re-done, or the settings for
contract payouts in the pre-shipped difficulty settings are going to have
to be edited. This is going to require some extensive playtesting to get a
proper balance.

**Comment by aw1621107** [comment]
@Rockstar04 I think KCT sort of does that indirectly, by preventing you from placing Kerbals into things they haven't been trained for (or at least that's how I remember it working). A more explicit warning would be nice, though.

I think the training contract could be tricky, because some players may go for the sound barrier contract early on, and that already requires training. A relatively small contract sounds like an interesting possibility to me. It can be available early on, to ensure the player gets at least somewhat acquainted with the training system before the real fun starts. A fairly small reward would help avoid breaking early game balance.

@pap1723 I agree w.r.t tooling vs the other features. Maybe we should at least get a wish list of the yet-to-be-completed features together, so we can hack at it a little at a time while NK gets some R&R. Incremental progress is better than no progress.

Does training and maintenance have that much of an effect on your funds? I'm all for not including them in this release, but idk how much rebalancing they would require.

I'm not sure rollout costs need to be removed completely; I think it's the tags that would pose a problem, since the multipliers for those are invisible. Otherwise, I think rollout costs don't add anything too complex to gameplay, and has a pretty sensible explanation (most rocket costs are in assembly/prep, not just part costs). The only missing feature I feel might make rollout costs not ready for release is that there's no way to break down the rollout costs to see what portion comes from where. That is already the case with build points, though, so I don't think it'd be that problematic.

I agree that balancing for the missing multipliers could be tricky, especially later in a playthrough when more expensive ones start coming up. Would it be possible to write a script that manually applies them? Basically `if HAS(tag) then cost *= $tag` (or the equivalent for how tag multipliers work).

**Comment by Schnobs** [comment]
It took me a good long while to even notice that there are maintenance costs. These, and training, certainly don't require that all contract payouts need to be recast.

I play without rollout (simply because I hate how it drops me out of warp just so I can press a button), so I can't talk about that. I hear that it's rather expensive. But in the end, no rollout costs only means that you have more money to pipe into upgrades -- from what I gather, on IRC, reddit and so on, people are progressing at vastly different timescales. Some manage Mercury by 1955, others not until the 70s. I'm inclined to say that there s no "right" funding anyway, so, no worries.

**Comment by Rockstar04** [comment]
@aw1621107 
> I think KCT sort of does that indirectly, by preventing you from placing Kerbals into things they haven't been trained for (or at least that's how I remember it working). A more explicit warning would be nice, though.

This is correct, but it was not obvious to me at first why I could not fill the pod, and even after doing proficiency and mission training there were still a few times that I could not get KCT to fill the pod so I had to hire and train a completely new Kerbal. I think we all agree here, that better communication to the player WHY you cant fill a pod would be very helpful.

**Comment by aw1621107** [comment]
> simply because I hate how it drops me out of warp just so I can press a button

idk whether this is a KCT or KAC thing. If it's KCT giving KAC an alarm , it should be a matter of making the alarm a do-nothing alarm (if those exist?) instead of pause-warp alarm.

> I hear that it's rather expensive.

From what I remember about my last playthrough it's usually a few times more expensive than the rocket itself for "normally-sized" rockets (i.e. not sounding rockets). 

> I'm inclined to say that there s no "right" funding anyway, so, no worries.

I agree that there's no "right" funding, especially since there are so many playstyles supported. Might a good balance be figuring out how most newer players play, and balancing for that, assuming that more experienced players have the knowledge to pick good settings themselves? Maybe providing a way to share difficulties for different playstyles would make that a bit more acceptable.

**Comment by aw1621107** [comment]
@Rockstar04 

> even after doing proficiency and mission training there were still a few times that I could not get KCT to fill the pod so I had to hire and train a completely new Kerbal.

I've experienced something like that too. There's supposedly been fixes to training sometime relatively recently (past few months?), but I haven't gotten to play with it myself. Open an issue if it happens again?

**Comment by Rockstar04** [comment]
There are already several discussions here on github that are covering rollout costs specifically. Personally I find it odd that "rollout" costs are higher than the rockets total cost.

#856 - Launchpad sizing, launcher-classes and rollout costs
#771 - Manned flight costing issues?
#719 - Adjusting RP0 costs to reflect integration prices, and overall KCT formula tweaks [discussion]

**Comment by Schnobs** [comment]
> Personally I find it odd that "rollout" costs are higher than the rockets total cost.

Good discussion in #719, but here's my attempt at a very short answer: according to the KSP editor, the vessel price is the sum of all part prices: Putting them together has no cost. "Rollout" is supposed to account for integration, assembly, checkout, launch services, mission control, what have you.

**Comment by aw1621107** [comment]
> There are already several discussions here on github that are covering rollout costs specifically

I think #771 is more a problem with how tags interacted with other bits of RP-0 than a rollout thing. The other two are definitely issues, but I think they're also part of a larger decision that needs to be made about whether and how all these wildly divergent playstyles can be accommodated. We could release with rollout costs as is, because that's what the playtesters so far are going to be most familiar with, and we can use the number of upse^H^H^H^Hhappy users to gauge whether that was a good or bad decision.

I think another part of the problem is that there may be an "ideal" rollout implementation, but we may not have the time/manpower to get that working, especially if we need to start touching other mods. Thus, we get to try to make things work with what is already available.

**Comment by DegenerateGroundState** [comment]
I'm only a random user but I'd take half-implemented confusing new features every time over a simplified release, if I wanted simple I could just play vanilla.

**Comment by pap1723** [comment]
There is nothing simple about what would be released. Half implemented
features take away from gameplay in my opinion.

**Comment by ec429** [comment]
Summary: I'm OK with the "tool it all" button and the existence of an option to turn off training (I would keep it on, but I can see why others wouldn't).  The rest, though, you can pry the half-implemented features from my cold dead hands.  I say we release RP-1 with all the current features still in.

 * Maintenance cost.  People are saying "this is ignorable"; if so, feel free to ignore it.  I don't see a reason there for turning it off.
 * Tooling.  Adds so much to gameplay and definitely affects balance; with a "tool all" button there's really nothing left to complain about.  Plus there are complicated questions about what exactly a toggle would do to a running game, so let's not go down that road.
 * Rollout.  IMNSHO there is absolutely nothing wrong with what we have.  However, I've not gone beyond the 350t pad, so it's possible the cost grows too much on the highest-level pads; but I'm sure we can tweak that later.
 * Training.  While I think all the bugs got fixed, I agree the interface isn't very user friendly so I accept people may want an option to turn it off and not have to worry about all that.  The effects on gameplay really only matter if you're racing.
 * Crew science.  It basically works, but _definitely_ need something in the Readme telling users how to turn on their CO₂ scrubbers.  Wild Blue makes the existing PAW button disappear and you have to find it in the 'Manage Operations' GUI, where it doesn't have a label (it's just "there is a converter in here, turn it on").  TBH if people wanted to remove this one for now I'd be OK with that; the amount of science involved is small enough that it doesn't really affect balance, and you still have enough incentives from contracts for manned missions to be worthwhile (I find that by the time I get that far, I'm always swimming in science points anyway).
 * Probe science.  This is really well-made, well done @pap1723 !  AFAIK there are no issues here at all.
 * New facilities levels.  There's a small issue here that the 75-point R&D upgrade only unlocks a handful of techs, many of which are empty or contain only non-configured parts.  But that's definitely not a release blocker.

**Comment by NoMrBond** [comment]
To complement the/a "Tool All" button, a "Set To Tooled" button/option would be useful.  e.g. when adding a pTank an option to quickly set it to an existing tooling option for that tank type, so when adding a Tank-III you have the option to snap it to an already purchased tooling the player owns for Tank-III.

**Comment by aw1621107** [comment]
> Maintenance cost. People are saying "this is ignorable"; if so, feel free to ignore it. I don't see a reason there for turning it off.

In this particular case, I was thinking that it would be more for a smoother/easier release than because the feature is half-baked. I figured that splitting the release into mandatory/big changes (tooling and new science/tech tree at least) first, then optional/smaller changes + bugfixes later could make things a bit easier both for the players and for customer support -- take care of the changes that are more likely to confuse people first, then once they're used to it drop smaller/optional features.

I haven't done releases before, though, so I have no clue how good/bad of an idea this is.

> Rollout. IMNSHO there is absolutely nothing wrong with what we have. However, I've not gone beyond the 350t pad, so it's possible the cost grows too much on the highest-level pads; but I'm sure we can tweak that later.

I think the mechanic is fine, it's just that tag multipliers have horrendously bad discoverability. Maybe adding something to the module description like "Increases build costs by 50%" would help? Basically some way to help players decide whether a more complex part is worth the extra build costs (e.g. turbopump liquid engines are 4x, but pressure-fed are 1.75x, so there could be significant savings for using a less efficient pressure-fed engine).

> a "Set To Tooled" button/option would be useful.

Interesting idea. Maybe even better would be allowing the player to click on a tooling entry to use that size? That way you don't have to get the tank close to the desired dimensions before snapping, especially if tooling dimensions are closely spaced.

**Comment by pap1723** [comment]
I agree with a lot of what is said on here, here is my proposal, shoot it
down, modify it, or make one of your own and we will all decide how to move
forward.

Release RP-1 for 1.3 that includes the following changes:

   - Overhauled Tech Tree
   - New Science System
   - Tooling
   - Building Tiers
   - Crew Science
      - For me, it is a nice thing that allows you to have another reason
      to launch crewed missions.
   - Rollout Costs
      - The caveat here is that we need to find a way to explain to the
      player what costs are increased. This can easily be done by adding a MM
      that attaches a description to any part that has a specific Module tag
      attached to it. Someone will just need to figure out exactly what those
      modifiers are, and if they make sense.
   - Strong focus on a proper Readme / Wiki. This is something that has
   been severely lacking for RP-0 / RO for basically ever. I know that when I
   was trying to get into RO, there was nothing I could do to learn it. The
   first two times I decided I was going to try it, I quit and went back to
   stock-alike because I didn't know what was going on. Now, I am not saying
   we need to have a fully fleshed out Wiki before release, but at a minimum,
   the tech tree, Science Experiments, Crew Science, Tooling, Building Tiers
   and Rollout Costs need to be explained.

Do not include and the reasons why...

   - Maintenance Costs
      - As many have stated, it is a negligible cost and is not usually of
      a concern for the player. To me, this is one of those features
that should
      not be included because there is not a good reason for having it
in. Sure,
      there are maintenance costs in real life, but until we can
design a fleshed
      out system that gives the player visibility on what is increasing their
      maintenance costs, or how to control them, it should not be included just
      to be in. Look into adding this as a redesigned feature in an upcoming
      release.
   - Training
      - The idea behind training is great, but I do not think it is
      implemented in a player friendly way yet. It should be re-designed to be
      more in line with Buzz Aldrin RIS style where you have them train for
      programs, can pull them out, get notices, etc. Look at adding this in a
      re-designed way in the future.

Just my 2 cents, so I am up for more discussion / agreement / disagreement
to flesh out the best possible route forward.

**Comment by Schnobs** [comment]
> Maintenance cost. People are saying "this is ignorable"; if so, feel free to ignore it. I don't see a reason there for turning it off.

Neither do I. The RP-0 window has several tabs dedicated to Maintenance cost and how it's made up.  I see no problem with the mechanism as such. That it's so cheap as to go virtually unnoticed might be a balance issue, but is another reason why I don't think it's going to bite anyone. A master switch to toggle it might be nice anyway, but is certainly no prerequisite for a RP-1 release.

Rollout, OTOH... Though the price may be right, it seems utterly ridiculous unless you know what it's supposed to represent. "How can rollout be X times as expensive as the rocket" is shaping up to become _the_ FAQ. I'd say it should not be unleashed on the public at large in it's current form. I like aw1621107's idea of adding notes to the part descriptions, but am pretty sure that it won't suffice.

Crew science: Interesting. I didn't know about the converter issue, and was looking for TACLS bugs and configuration errors. No need to rip it out of RP-1, though: "Removal" could be achieved by simply not recommending/suggesting it on CKAN, so only people who check the Wiki will learn about it. Who will hopefully also pick up on the caveats.

**Comment by aw1621107** [comment]
After thinking a bit more, I'm leaning towards not including maintenance costs. I think that they have the potential to add interesting gameplay, but definitely not in its current form. We should add a way for players to reduce maintenance costs (FTL power management may be an interesting starting point) and ideally there would be something that gives an estimate of how much time the player has left before running out of funds, but that's not quite as important.

So I guess I feel maintenance costs are in the same category as training -- good idea, wanting execution.

But besides those two, I think that the other stuff should be included, modulo bugs and balancing fixes.

By the way, what do you guys think about how avionics work at the moment? I think they're usable, but still need some polishing.

**Comment by Brummi1234** [comment]
@aw1621107 
I totally agree with your ideas.

Regarding avionics, i think the concept is good as it is.
There is just one bug(?), if a part is not listed in the avionics.cfg but has crew capacity, it gets avionics in game, but without mass restriction.
The cause may be the toggle/shutdown that gets added to all the crewed parts.
Would be nice if this could somehow be fixed or worked around.

**Comment by NoMrBond** [comment]
> Interesting idea. Maybe even better would be allowing the player to click on a tooling entry to use that size? That way you don't have to get the tank close to the desired dimensions before snapping, especially if tooling dimensions are closely spaced.

Either being able to "pick up" a tank from the tooling list by clicking on its entry sounds good.  Being able to convert/shape/snap a selected pTank by clicking on an entry in the tooling list might also be useful but yes the first one sounds best.

**Comment by Bornholio** [comment]
lots of suggestion of additions. realistically unless your willing to add and test those they are not happening.
Awang has already done a cool tool all button so that is pretty likely.

As for roll out, my perception of it was that an atlas-mercury launch is as expensive or more after i add some solar power as a shuttle launch adjusted for inflation.  I imagine that building and launching a full shuttle will be program breaking.  I just doesn't have balance done for the higher end.

of course if you take any feature out it won't get tested and balanced 📦

**Comment by Schnobs** [comment]
> of course if you take any feature out it won't get tested and balanced

I guess in it's current form you can't expect much better feedback than we have already. It's a lump sum that first becomes visible some time after you've left the VAB. Pretty hard for a player to figure it out, even if he tries to.

**Comment by ec429** [comment]
> a lump sum that first becomes visible some time after you've left the VAB
It should be visible in the VAB, in the KCT GUI.

**Comment by Rockstar04** [comment]
>> a lump sum that first becomes visible some time after you've left the VAB
>
> It should be visible in the VAB, in the KCT GUI.

Rollout costs are visible alone with the estimated build time

**Comment by aw1621107** [comment]
> There is just one bug(?), if a part is not listed in the avionics.cfg but has crew capacity, it gets avionics in game, but without mass restriction.

@Brummi1234 Can you open an issue for this? It sounds like it should be fast, but I don't want to forget about it the next time I boot up KSP things

> Being able to "pick up" a tank from the tooling list by clicking on its entry sounds good.

Well, that wasn't really what I was thinking... That basically sounds like re-implementing the part picker (if that's even possible), and that's definitely something I'd prefer to avoid. I was more thinking about having a list of toolings, and doing something that lets you select a proc part then apply the tooling from the list.

This fun functionality may not be fun to implement. It all comes down to the game needing to know which proc part you want to change. My best idea at the moment is to implement something similar to that mod that lets you center your camera on different bits of your vessel, so you hover over the part, press a key, then click the size you want. There's clicking on the size while dragging the part, but I'm not sure how much people like click-through. Changing the "last clicked" proc part is also another option, but I'm not sure how easy that would be to implement. The first choice at least has the advantage of being in at least one other mod.

> of course if you take any feature out it won't get tested and balanced

I think it's more "How closely will the final product match what currently exists?". If there's going to be a huge UI/UX change, then releasing may not be the best idea; don't want to basically remove an old feature and add a new one that shares the name of the old one. If the feature is going to be the "same", for an extremely loose definition of "same", then I think it should go out.

**Comment by Schnobs** [comment]
> Rollout costs are visible alone with the estimated build time

That's what I get for turning it off completely (which is easy: the KCT setting for reconditioning toggles rollout as well, hence bypassing rollout cost.). Still, just one figure -- unlike with a sausage, I'd really like to know how it's made.

**Comment by ockidj** [comment]
Maybe we could have tooling, training, and maintenance in the save file difficulty options? With tooling being the only one toggled by default.

**Comment by NathanKell** [comment]
Hi all! Chiming in, finally. :)

First, let me offer both a realism and a gameplay explanation for each of maintenance costs and training. IMO the gameplay side is the more relevant in both cases.

Maintenance costs:  they should be a nontrivial fraction of the space program's yearly budget (realism), which means they probably need to be turned up a bit to really feel them. Gameplay-wise, the reason for them is to prevent just warping until all tech is done, i.e. to give the player some pressure. Otherwise there's really no reason you can't just queue up a bunch of tech, wait 10 years, and then resume your space program. It forces (or should force) the player to play at every stage of development and career, and not sit idly by.

Training: The realism aspect here is quite obvious, given how much we've lionized astronaut training (and astronauts period). But the gameplay side is why I started working on it: it forces you to *plan*. You can't undertake a mission on a whim. Human spaceflight should feel like a commitment, not a whim, and the biggest concern for it, not even money, is *time*. Time is precious (see maintenance above), and HSF is a commitment on not just the monetary level. It also forces you, IMO interestingly, to be using older hardware than you otherwise would.

Now that that's out of the way, I strongly suggest leaving maintenance costs in and tuning them. It also should not be terribly hard to add a "downgrade" button to KCT, just as there's a built-in KSP way to downgrade facilities. Since we already bind to KCT we could do it on our side, actually, if we had to.

Regarding training, per above I would definitely plop for including them, with a little bit of usability. It should be fairly simple to add a popup when you unlock a crewed part that says "remember to train your nauts" -- there's already an event handler for "part got unlocked" that handles the entry cost stuff, so we can just also spawn a popup if crewcapacity > 0. It should also be fairly simple to add KAC alarms, can just steal the KCT code since KCT already does it. Finally, for the warning that no crew are trained, that would require KCT-side changes I believe. One thing we can do is highlight the vessel name in red in the build list if no crew can be placed in any of its parts. We can also (as I should have in the first place) add a note to the launch dialog if there's no proper crew. And we should *definitely* edit the contract text for all HSF missions to say "remember to train crew for this mission!"

It is, however, much more of a "frustrator" feature than the others, I acknowledge. Most of the other stuff just makes doing things slower or more expensive, but forgetting training means you can't do what you want *at all*, and not even necessarily know why at the start, and that can indeed be frustrating. I feel that adding UI feedback can at least make clear what's wrong, but it won't fix the "oh crap I'm screwed, I forgot to train" part of the frustration.
But then RO is all about not forgetting. :)

As to rollout costs: IMO they are very, very necessary. Noting on part descriptions what the multipliers are might ease the pain for sure, but, at least by me, if we want realistic part costs, we need launch costs. If we want to skip launch costs, we absolutely need to rejigger (and un-realism) part costs, because otherwise, as @Schnobs has often pointed out, why would you ever use solids and not hydrolox. And we'd need to bake in the cost of crewed missions some other way. So...yeah. Needs better UI, sure, but even with the UI it's got it's (gameplay-wise) better than not having it, by far.
And it combos very well with tooling and reusing existing LVs.

**Comment by NathanKell** [comment]
Oh! Right! Avionics.
Avionics is "opt-in". That means that any ModuleCommand that *doesn't* have a ModuleAvionics sitting beside it gives unlimited control. That's why if you add a command part that we haven't avionics-ized, you don't have to worry about avionics.

**Comment by NathanKell** [comment]
Final note: I understand that I totally didn't finish training though, and can see the argument for nixing it for now. I just tend to feel like, for a big enough change where we even change the name of the thing (or at any rate break saves) we might as well have everyone relearn everything *once*, even if it's a bit sharp-edged. :)

**Comment by glilienthal** [comment]
I've been playing RP1 for a while now. Great step forward. Lot's things I like,  some I don't mind, but one is really messing up my gameplay: Retirement. 

I find the rules for retirement obscure and the UI not that easy. I am  never sure what brings retirement and what delays it.

I'd suggest an option to opt out of retirement in the difficulty settings.

**Comment by glilienthal** [comment]
Re the rollout costs: Make sense to me. I found it always sad that using solids was not really helping my spaceprogram. At some point in time, however, it'd be really helpful to know which part "produces" how much rollout costs.

**Comment by Bornholio** [comment]
I'm willing to say that its worth releasing as is at 1.3.1 level, and then any forward changes focus on 1.4.x
Anything broken isn't unplayable so, that said a few infrastructure items for CKAN need updating and pushing.

**Comment by Relic13** [comment]
Greetings all.  So, for what it's worth, my 2cents on rp-1.  Overall, I think quite a lot is great, some things are rough/unrefined (but it is to be expected) and some are really flaws for game play IMO.
I love the implementation of payloads, seems similar to what I had suggest to nathankell and pap1723 a while back, and the science rework is even better!

I guess only two main issues of implementation for gameplay that I see, being rollout costs and naut training.

Also, maintenance is trivial currently and should probably be more noticeable if kept.  However, it currently doesn't seem like the player has any thoughtful decisions to make regarding maintenance cost - they just are what they are... So unless there is some player interaction into really influencing maintenance costs with interesting decisions, couldn't the point of a pure money sink to make people launch more often instead just as easily be accomplished by reducing the payouts? - so they have to launch more contract missions to earn the same amount.  I mean, isn't that what hard is for anyway? that you _have_ to launch contract missions all the time to be able to afford new stuff?

Anyway, First rollout costs.
I get the problem that it is addressing regarding total launch costs, and indeed there also should be a real incentive for using solids etc.... But doing this by adding rollout costs and hidden multipliers largely serves to only further obfuscate the whole thing for the player, and it adds a lot of complexity where it isn't needed at all.  IMO it would be much better to tweak the costs that are shown in the VAB parts themselves.  

Yes, the part costs largely have sources for the production cost of that individual part, however it should also be assumed that the player is not interested in buying 1 part just for having it sitting in a garage.... They are always going to be buying parts to assemble them into something and launch it on a mission (or scrapping launch which can refund the portion anyway)... So why not just include these assembly costs into the shown part costs themselves instead of obfuscating the information?  It would make the information far easier for the player to discern.

IE just rejigger the part cost value you see in the VAB to be the cost of part + its inferred assembly cost.  This way players can easily see the financial impact each part has/would have on costs for the rocket/probe they are building - and it is a huge quality of life improvement so that players who are choosing a craft for a mission from their saved list will see the actual cost without having to click into load each vessel to remember the additional rollouts to compare.  (For example I often will save launch vehicles by payload tonnage to LEO, so when I look at the list of saved craft it would be incredibly nice if the cost displayed next to each craft name was the total cost of using that craft instead of having to open every similar tonnage vessel to remember to add and compare the roll out costs).

And second, astronaut training. 
What I do think is great for game play, and I think what NathanKell was getting at with initial training is that nauts require a time commitment for new capsules, between missions and before launches.  This is pretty cool since it does add a new dimension of gameplay to consider that is unique to manned missions.
However, the implantation is IMHO quite bad for game play.  I'm going to be deliberately facetious here: the way proficiency / mission training / refreshers / retirement is done comes off is a lot like "hey space agency's have a payroll system, lets add that for realism.  you click a button every month to pay your nauts or they'll quit on you, also don't forget to click to give them hazard pay before a mission or they'll refuse to do that too.  just clicking for the sake of clicking...

IMO the whole naut training thing could be far more simplified while retaining all the good game play elements of time commitments without the added player frustration of having to periodically navigate menus just to click buttons over and over.

Something like:
- Keep initial training for each capsule type (and probably up the time of training for this).
- Have much more Naut downtime after recovery (imply refreshers happen in this down time and some future mission training). 
- And if you want to have it so a player can't just immediately launch a manned mission on a whim, then maybe have a minimum roll out time if a manned pod is on it. (ie imply further safety checks and some mission training).

This way the good game play elements of time commitment for manned missions are retained for the player, and the massive nuisance of having to deal with the tedium of clicking and constantly checking through menus of when/who to click again is almost entirely removed.

Cheers!
Relic.

**Comment by kurgut** [comment]
Hi, I freshly installed RP-1, and this is great, work fine, thank you guys for bringing that stuff! :)

I don't know if it is the correct place to suggest, but anyway:

I think maybe the KCT build time should decrease when you build a rocket that includes parts that you have arleady built before, just like in real life: building a engine for the first time takes more time to build than the second time, etc. So there should be a decreasing build time for those already builded parts going to a minimum limit to stay realistic.

Thank you :)

**Comment by glilienthal** [comment]
I second everything Relic13 wrote. - I have been logging a few tens of hours on that, and those were - IMHO - the best options to improve gameplay.

**Comment by pap1723** [comment]
RELEASED 548e4ac134c0253ec6dfecbcefcb60dac64e5240

**Comment by pap1723** [comment]
@Relic13 we are definitely looking into overhaul of the entire Crew system
